### DISCUSSION TITLE
Experimental major release 7.0

### DISCUSSION THREAD

**Discussion by jamesdaniels** [angular/angularfire#2713] [Roadmap]
**This is a placeholder for the discussion of the vNext experimental branch of AngularFire (`7.0`) which provides first-class support for `firebase@beta`**.

### Background

TBD

### Goals

* Support `firebase@beta`
* Allow for maximum use of the tree shaking benefits within Angular applications
* Revisit dynamic imports, promise proxy, and other patterns that were working around large bundle-sizes
* Rewrite the AngularFire API to be fully tree-shakable itself

### Non-goals

* Supporting `firebase@beta` with the current (`6.1`) API

### More information

* [`firebase@beta` docs](https://github.com/firebase/firebase-js-sdk/blob/master/docs-exp/index.md)
* [Public Roadmap card](https://github.com/angular/angularfire/projects/5#card-50953968)
* [Milestone](https://github.com/angular/angularfire/milestone/24)

**Comment by DMZakaria** [comment]
hi, @jamesdaniels thanks for your hard work on angular fire. Do you have an official release date for version 7?
Because I don't see the date on the roadmap

**Reply by jamesdaniels** [reply]
It’s our policy not to have public time tables. But we’re aiming for AngularFire v7 to go GA alongside Firebase v9. I’m working on getting started guides for the next release now.

**Reply by BCusack** [reply]
With the Firebase v9 release, would this be a major re-write for the Angularfire docs? so many breaking changes...

**Reply by jamesdaniels** [reply]
We opted to keep most the docs using `@angular/fire/compat/*` for now and will be migrating as we bring the new API up to partity.
